### PR TITLE
Master hr attendance close7days vlo

### PR DESCRIPTION
**Pull request by LofbergValentin**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
@LofbergValentin please do not forget to add a link to the task and the URL to the read-only version of the pad

File: addons/hr_attendance/static/src/js/kiosk_mode.js [hunk 1]
```diff
@@ -55,9 +56,20 @@ var KioskMode = AbstractAction.extend({
...
         this._super.apply(this, arguments);
     },
+
+    call2Db: function () { 
+        return this._rpc({
+               model: 'res.users',
+               method: 'search_read',
+           })
+           .then(function (users){
+               console.log(users[0].id);
```
Comment (tivisse) [inline addons/hr_attendance/static/src/js/kiosk_mode.js]: @LofbergValentin Is this a remaining debugging chunk of code ?

Comment (LofbergValentin) [reply addons/hr_attendance/static/src/js/kiosk_mode.js]: You are right fixing that right away

File: addons/hr_attendance/static/src/js/kiosk_mode.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ var KioskMode = AbstractAction.extend({
                 self.$el.html(QWeb.render("HrAttendanceKioskMode", {widget: self}));
                 self.start_clock();
             });
+        this._interval = window.setInterval(this.call2Db.bind(this), (((1000 * 60) * 60) * 24));
```
Comment (ged-odoo) [inline addons/hr_attendance/static/src/js/kiosk_mode.js]: no need for the many parenthesis: you can simply write 1000*60*60*24

File: addons/hr_attendance/static/src/js/kiosk_mode.js [hunk 3]
```diff
@@ -55,9 +56,17 @@ var KioskMode = AbstractAction.extend({
 
     destroy: function () {
         core.bus.off('barcode_scanned', this, this._onBarcodeScanned);
+        clearInterval(this._interval);
         clearInterval(this.clock_start);
         this._super.apply(this, arguments);
     },
+
+    call2Db: function () {
```
Comment (ged-odoo) [inline addons/hr_attendance/static/src/js/kiosk_mode.js]: this should be a private function, and with a better name. For example: _callServer, also, it needs documentation, which explains why this is needed

File: addons/hr_attendance/static/src/js/kiosk_mode.js [hunk 4]
```diff
@@ -55,9 +56,17 @@ var KioskMode = AbstractAction.extend({
...
+        return this._rpc({
+               model: 'res.users',
+               method: 'search_read',
```
Comment (ged-odoo) [inline addons/hr_attendance/static/src/js/kiosk_mode.js]: usually, for this scenario, we use the route '/web/webclient/version_info'

Comment (LofbergValentin) [reply addons/hr_attendance/static/src/js/kiosk_mode.js]: I need it in the hr_atttendance module wich does not depends on web

Comment (tivisse) [reply addons/hr_attendance/static/src/js/kiosk_mode.js]: Actually ``hr_attendance`` depends on ``hr`` that depends on ``web``.

**Comment by tivisse** [comment]
This branch should target the 10.0. 

The work continues here: https://github.com/odoo/odoo/pull/24898/commits

**Pull request closed without merging**